dcos-commons/src/main/java/org/apache/mesos/offer/DefaultOfferRequirementProvider.java

```diff
@@ -155,4 +157,23 @@ private void validateVolumes(Protos.TaskInfo taskInfo, TaskSpecification taskSpe
 
         return volumes;
     }
+
+    private Protos.ExecutorInfo getExecutorInfo(TaskSpecification taskSpecification) {
+
+        Protos.ExecutorInfo.Builder executorInfoBuilder = Protos.ExecutorInfo.newBuilder()
+                .setName(taskSpecification.getName())
+                .setExecutorId(Protos.ExecutorID.newBuilder().setValue("").build()); // Set later by ExecutorRequirement
+
+        Protos.CommandInfo.Builder commandInfoOrBuilder = Protos.CommandInfo.newBuilder()
+                .setValue("./custom-executor/bin/custom-executor")
+                .addUris(TaskUtils.uri("https://s3-us-west-2.amazonaws.com/infinity-artifacts/custom-executor.zip"))
```

Comment:
Shouldn't this URI instead be provided via `resource.json` in packaging? Otherwise this will break in airgapped clusters.